### ISSUE TITLE
cannot import name 'resnet '

### ISSUE THREAD

**Issue by zihaozhang9** [tensorflow/tensorflow#36371]
I installed tensorflow using the following command
`conda install tensorflow-gpu`
`conda install keras-gpu`
then
`import tensorflow`
I get the following error:
>>> import tensorflow
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow/__init__.py", line 98, in <module>
    from tensorflow_core import *
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/__init__.py", line 40, in <module>
    from tensorflow.python.tools import module_util as _module_util
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow/__init__.py", line 50, in __getattr__
    module = self._load()
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow/__init__.py", line 44, in _load
    module = _importlib.import_module(self.__name__)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/__init__.py", line 83, in <module>
    from tensorflow.python import keras
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/__init__.py", line 26, in <module>
    from tensorflow.python.keras import activations
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/__init__.py", line 27, in <module>
    from tensorflow.python.keras import applications
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/applications/__init__.py", line 64, in <module>
    from tensorflow.python.keras.applications.resnet import ResNet50
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/applications/resnet.py", line 22, in <module>
    from keras_applications import resnet
ImportError: cannot import name 'resnet'

**System information**
-ubuntu18.04

anaconda env:
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main    defaults
_tflow_select             2.1.0                       gpu    defaults
absl-py                   0.8.1                    py36_0    defaults
astor                     0.8.0                    py36_0    defaults 
blas                      1.0                         mkl    defaults
c-ares                    1.15.0            h7b6447c_1001    defaults
ca-certificates           2020.1.1                      0    defaults
certifi                   2019.11.28               py36_0    defaults                                               
cudatoolkit               10.0.130                      0    defaults
cudnn                     7.6.5                cuda10.0_0    defaults                                                    
cupti                     10.0.130                      0    defaults
gast                      0.2.2                    py36_0    defaults                                                  
google-pasta              0.1.8                      py_0    defaults
grpcio                    1.16.1           py36hf8bcb03_1    defaults                                            
h5py                      2.10.0           py36h7918eee_0    defaults
hdf5                      1.10.4               hb1b8bf9_0    defaults               
intel-openmp              2019.4                      243    defaults
keras-applications        1.0.8                      py_0    defaults                                                           
keras-base                2.2.4                    py36_0    defaults
keras-gpu                 2.2.4                         0    defaults                                                                 
keras-preprocessing       1.1.0                      py_1    defaults
ld_impl_linux-64          2.33.1               h53a641e_7    defaults                                                                 
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults                                                                              
libgcc-ng                 9.1.0                hdf63c60_0    defaults
libgfortran-ng            7.3.0                hdf63c60_0    defaults                                                                            
libprotobuf               3.11.2               hd408876_0    defaults
libstdcxx-ng              9.1.0                hdf63c60_0    defaults
markdown                  3.1.1                    py36_0    defaults
mkl                       2019.4                      243    defaults
mkl-service               2.3.0            py36he904b0f_0    defaults
mkl_fft                   1.0.15           py36ha843d7b_0    defaults
mkl_random                1.1.0            py36hd6b4f25_0    defaults
ncurses                   6.1                  he6710b0_1    defaults
numpy                     1.18.1           py36h4f9e942_0    defaults
numpy-base                1.18.1           py36hde5b4d6_1    defaults
openssl                   1.1.1d               h7b6447c_3    defaults
opt_einsum                3.1.0                      py_0    defaults
pip                       20.0.2                   py36_1    defaults
protobuf                  3.11.2           py36he6710b0_0    defaults
python                    3.6.10               h0371630_0    defaults
pyyaml                    5.2              py36h7b6447c_0    defaults
readline                  7.0                  h7b6447c_5    defaults
scipy                     1.3.2            py36h7c811a0_0    defaults
setuptools                45.1.0                   py36_0    defaults
six                       1.14.0                   py36_0    defaults
sqlite                    3.30.1               h7b6447c_0    defaults
tensorboard               2.0.0              pyhb38c66f_1    defaults
tensorflow                2.0.0           gpu_py36h6b29c10_0    defaults
tensorflow-base           2.0.0           gpu_py36h0ec5d1f_0    defaults
tensorflow-estimator      2.0.0              pyh2649769_0    defaults
tensorflow-gpu            2.0.0                h0d30ee6_0    defaults
termcolor                 1.1.0                    py36_1    defaults
tk                        8.6.8                hbc83047_0    defaults
werkzeug                  0.16.1                     py_0    defaults
wheel                     0.34.1                   py36_0    defaults
wrapt                     1.11.2           py36h7b6447c_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zlib                      1.2.11               h7b6447c_3    defaults
(tfenv) nnir712@nnir712-Lenovo:~/temp$ python
Python 3.6.10 |Anaconda, Inc.| (default, Jan  7 2020, 21:14:29)

**Comment by wdirons** [comment]
I would try `conda remove keras-gpu`

As TensorFlow 2.0 comes with tf.keras, you don't want to use the keras conda package. (Which at version 2.2.4 only works with TensorFlow 1.X)

**Comment by zihaozhang9** [comment]
> I would try `conda remove keras-gpu`
> 
> As TensorFlow 2.0 comes with tf.keras, you don't want to use the keras conda package. (Which at version 2.2.4 only works with TensorFlow 1.X)

I will use keras.then. use  `pip install keras==2.0.8` ?

**Comment by wdirons** [comment]
If using TensorFlow 2.0, you want to use tf.keras, all you need to install would be tensorflow-gpu.

https://www.tensorflow.org/guide/keras

**Comment by zihaozhang9** [comment]
> If using TensorFlow 2.0, you want to use tf.keras, all you need to install would be tensorflow-gpu.
> 
> https://www.tensorflow.org/guide/keras

ok.
 I am learning tensorflow. But some problems seem to appear. I try to run the [code ](https://www.tensorflow.org/tutorials/images/cnn). But found the following error

`Instructions for updating:
keep_dims is deprecated, use keepdims instead
Train on 50000 samples, validate on 10000 samples
Epoch 1/10
2020-02-01 06:34:31.435605: I tensorflow/stream_executor/cuda/cuda_gpu_executor.cc:898] successful NUMA node read from SysFS had negative value (-1), but there mu
st be at least one NUMA node, so returning NUMA node zero
2020-02-01 06:34:31.436650: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1356] Found device 0 with properties:
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.721
pciBusID: 0000:01:00.0
totalMemory: 10.92GiB freeMemory: 10.77GiB
2020-02-01 06:34:31.436675: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1435] Adding visible gpu devices: 0
2020-02-01 06:34:31.784720: I tensorflow/core/common_runtime/gpu/gpu_device.cc:923] Device interconnect StreamExecutor with strength 1 edge matrix:
2020-02-01 06:34:31.784761: I tensorflow/core/common_runtime/gpu/gpu_device.cc:929]      0
2020-02-01 06:34:31.784770: I tensorflow/core/common_runtime/gpu/gpu_device.cc:942] 0:   N
2020-02-01 06:34:31.785344: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1053] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with
10427 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:01:00.0, compute capability: 6.1)
 2976/50000 [>.............................] - ETA: 42s - loss: nan - acc: 0.99562020-02-01 06:34:34.000984: E tensorflow/stream_executor/cuda/cuda_driver.cc:1110
] could not synchronize on CUDA context: CUDA_ERROR_MISALIGNED_ADDRESS :: *** Begin stack trace ***
        tensorflow::CurrentStackTrace()
        perftools::gputools::cuda::CUDADriver::SynchronizeContext(perftools::gputools::cuda::CudaContext*)
        perftools::gputools::StreamExecutor::SynchronizeAllActivity()
        tensorflow::GPUUtil::SyncAll(tensorflow::Device*)
        tensorflow::BaseGPUDevice::Sync()



        Eigen::NonBlockingThreadPoolTempl<tensorflow::thread::EigenEnvironment>::WorkerLoop(int)
        std::_Function_handler<void (), tensorflow::thread::EigenEnvironment::CreateThread(std::function<void ()>)::{lambda()#1}>::_M_invoke(std::_Any_data const&
)
`

I'm not sure where the problem is. Will my graphics card hardware be damaged?

**Comment by wdirons** [comment]
TensorFlow with not damage a graphic card. [The nvidia software is responsible for slowing the card down or shutting it off so as to not overheat.]

I'm not sure about that error, but one think to check is to make sure you have the latest NVIDIA driver installed.

**Comment by zihaozhang9** [comment]
> TensorFlow with not damage a graphic card. [The nvidia software is responsible for slowing the card down or shutting it off so as to not overheat.]
> 
> I'm not sure about that error, but one thing to check is to make sure you have the latest NVIDIA driver installed.

Ok. thank you very much. I will reinstall the graphics driver, cuda and cudnn

**Comment by wdirons** [comment]
You only need to install the graphic driver. Because your using Conda, it will install cuda and cudnn as conda package (saving you work)   (For more info: https://stackoverflow.com/questions/59529804/nvidia-cudatoolkit-vs-conda-cudatoolkit)

**Comment by zihaozhang9** [comment]
> You only need to install the graphic driver. Because your using Conda, it will install cuda and cudnn as conda package (saving you work) (For more info: https://stackoverflow.com/questions/59529804/nvidia-cudatoolkit-vs-conda-cudatoolkit)

When I use conda install tensorflow-gpu
conda is installed:
  keras-applications pkgs/main/noarch::keras-applications-1.0.8-py_0                                                                                              
  keras-preprocessi~ pkgs/main/noarch::keras-preprocessing-1.1.0-py_1    
`import tensorflow`
Keep getting errors：ImportError: cannot import name 'resnet'
then
` pip uninstall keras-applications` 
` pip uninstall keras-preprocessing` 
`import tensorflow`
success!

But now it's a problem with the [code](https://www.tensorflow.org/tutorials/images/cnn)！
error：
`         [[{{node sequential/conv2d/Conv2D}}]]
   32/50000 [..............................] - ETA: 38:36Traceback (most recent call last):
  File "tfkeras_train_cifar.py", line 65, in <module>
    validation_data=(test_images, test_labels))
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training.py", line 728, in fit
    use_multiprocessing=use_multiprocessing)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2.py", line 324, in fit
    total_epochs=epochs)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2.py", line 123, in run_one_epoch
    batch_outs = execution_function(iterator)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/keras/engine/training_v2_utils.py", line 86, in execution_function
    distributed_function(input_fn))
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/def_function.py", line 457, in __call__
    result = self._call(*args, **kwds)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/def_function.py", line 520, in _call
    return self._stateless_fn(*args, **kwds)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/function.py", line 1823, in __call__
    return graph_function._filtered_call(args, kwargs)  # pylint: disable=protected-access
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/function.py", line 1141, in _filtered_call
    self.captured_inputs)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/function.py", line 1224, in _call_flat
    ctx, args, cancellation_manager=cancellation_manager)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/function.py", line 511, in call
    ctx=ctx)
  File "/home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/eager/execute.py", line 67, in quick_execute
    six.raise_from(core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InternalError:  cuDNN launch failure : input shape([32,3,32,32]) filter shape([3,3,3,32])
         [[node sequential/conv2d/Conv2D (defined at /home/nnir712/anaconda3/envs/tfenv/lib/python3.6/site-packages/tensorflow_core/python/framework/ops.py:1751)
]] [Op:__inference_distributed_function_1055]

Function call stack:
distributed_function`

code:
`
from __future__ import absolute_import, division, print_function, unicode_literals

import tensorflow as tf
#from tensorflow import keras
from tensorflow.keras import datasets, layers, models
from tensorflow.python.keras.datasets.cifar import load_batch
from tensorflow.python.keras import backend as K
#import matplotlib.pyplot as plt
import os
import numpy as np
def load_data():
  '''
  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/datasets/cifar10.py
  '''
  dirname = 'cifar-10-batches-py'
  path = os.path.join('/media/nnir712/I/opendataset',dirname)

  num_train_samples = 50000

  x_train = np.empty((num_train_samples, 3, 32, 32), dtype='uint8')
  y_train = np.empty((num_train_samples,), dtype='uint8')

  for i in range(1, 6):
    fpath = os.path.join(path, 'data_batch_' + str(i))
    (x_train[(i - 1) * 10000:i * 10000, :, :, :],
     y_train[(i - 1) * 10000:i * 10000]) = load_batch(fpath)

  fpath = os.path.join(path, 'test_batch')
  x_test, y_test = load_batch(fpath)

  y_train = np.reshape(y_train, (len(y_train), 1))
  y_test = np.reshape(y_test, (len(y_test), 1))

  if K.image_data_format() == 'channels_last':
    x_train = x_train.transpose(0, 2, 3, 1)
    x_test = x_test.transpose(0, 2, 3, 1)

  x_test = x_test.astype(x_train.dtype)
  y_test = y_test.astype(y_train.dtype)

  return (x_train, y_train), (x_test, y_test)
  
(train_images, train_labels), (test_images, test_labels) = load_data()
# Normalize pixel values to be between 0 and 1
train_images, test_images = train_images / 255.0, test_images / 255.0
class_names = ['airplane', 'automobile', 'bird', 'cat', 'deer',
               'dog', 'frog', 'horse', 'ship', 'truck']
model = models.Sequential()
model.add(layers.Conv2D(32, (3, 3), activation='relu', input_shape=(32, 32, 3)))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.add(layers.MaxPooling2D((2, 2)))
model.add(layers.Conv2D(64, (3, 3), activation='relu'))
model.summary()
model.add(layers.Flatten())
model.add(layers.Dense(64, activation='relu'))
model.add(layers.Dense(10, activation='softmax'))
model.summary()
model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

history = model.fit(train_images, train_labels, epochs=10, 
                    validation_data=(test_images, test_labels))
                    
test_loss, test_acc = model.evaluate(test_images,  test_labels, verbose=2)
print(test_acc)
`

**Comment by amahendrakar** [comment]
@zihaozhang9,
I was able to run the above code without any issues. Please find the Gist [here](https://colab.sandbox.google.com/gist/amahendrakar/7ff98a538316fe22ec11cc92ea7e3276/36371.ipynb). Thanks!

**Comment by zihaozhang9** [comment]
> @zihaozhang9,
> I was able to run the above code without any issues. Please find the Gist [here](https://colab.sandbox.google.com/gist/amahendrakar/7ff98a538316fe22ec11cc92ea7e3276/36371.ipynb). Thanks!
Thank you very much for your continued attention!
The day before yesterday, I tried many versions of graphics driver, CUDA, cudnn and tensorflow.
Then I discovered a magical phenomenon, when only 10 pictures are loaded in the graphics card, it can be trained normally. When loading 100 pictures, the video memory uses about 3G, and then reports an error.
From this I suspect that it is not the version that is wrong, it should be the graphics card hardware failure.I will contact nvidia's technical engineer to repair the graphics card.
